### PR TITLE
[1LP][RFR]Adding test case related to project.

### PR DESCRIPTION
**Pull request by anurag03**
Purpose or Intent
=================
Adding new test cases related to project quota.

{{pytest: -v -k 'test_project_quota_enforce_via_lifecycle_infra ' --long-running}}

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit e4f0a0ff07f81a525edc746ecee8b304f1acf161:

`cfme/tests/infrastructure/test_project_quota.py`:
- :bangbang: Line 96:1: **E302** *expected 2 blank lines, found 1*
- :bangbang: Line 118:45: **E127** *continuation line over-indented for visual indent*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit d6243ab079dac697e99b13cc969f1b633c9f95ef:

`cfme/services/service_catalogs/ui.py`:
- :bangbang: Line 76:14: **E126** *continuation line over-indented for hanging indent*

`cfme/tests/infrastructure/test_project_quota.py`:
- :red_circle: Line 10:1: **F401** *'cfme.utils.appliance.ViaUI' imported but unused*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 004bb38cd8c520d8e9f23c671fbe57fd78444b95:

`cfme/tests/cloud/test_project_quota.py`:
- :bangbang: Line 136:101: **E501** *line too long (103 > 100 characters)*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit b055e81c4c7c112346838df4c29ce680907b3369:

`cfme/tests/cloud/test_project_quota.py`:
- :bangbang: Line 136:101: **E501** *line too long (103 > 100 characters)*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit cb02ebf685c67110eca2554b494772d0fcf4a2e0:

`cfme/tests/infrastructure/test_project_quota.py`:
- :red_circle: Line 5:1: **F401** *'cfme.infrastructure.provider.rhevm.RHEVMProvider' imported but unused*
- :red_circle: Line 10:1: **F401** *'cfme.utils.appliance.ViaUI' imported but unused*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 7dc3bebfd5caa528bf45bf9d8bfbd52bdd9a9cfe:

`cfme/tests/infrastructure/test_project_quota.py`:
- :red_circle: Line 10:1: **F401** *'cfme.utils.appliance.ViaUI' imported but unused*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit a40fae1db7563a261860ea41cbbbf2ee26e90b6d:

`cfme/tests/infrastructure/test_project_quota.py`:
- :red_circle: Line 10:1: **F401** *'cfme.utils.appliance.ViaUI' imported but unused*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

File: cfme/tests/cloud/test_project_quota.py [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+    new_project.set_quota(**{'{}_cb'.format(field): False})
+
+
+@pytest.fixture(scope='module')
+def new_project(appliance):
+    collection = appliance.collections.projects
+    yield collection.create(name='project{}'.format(fauxfactory.gen_alphanumeric()),
+                            description='project_des{}'.format(fauxfactory.gen_alphanumeric()),
+                            parent=collection.get_root_tenant())
+
```
Comment (rlbabyuk) [inline cfme/tests/cloud/test_project_quota.py]: Required: Add project cleanup here

File: cfme/tests/cloud/test_project_quota.py [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+
+@pytest.yield_fixture(scope='module')
```
Comment (rlbabyuk) [inline cfme/tests/cloud/test_project_quota.py]: Required: yield_fixture is not mandatory to use now as @pytest.fixture in new version of pytest do the same

File: cfme/tests/cloud/test_project_quota.py [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+def new_group(appliance, new_project):
+    collection = appliance.collections.groups
+    return collection.create(description='group_{}'.format(fauxfactory.gen_alphanumeric()),
```
Comment (rlbabyuk) [inline cfme/tests/cloud/test_project_quota.py]: Required: use yield, and add cleanup

File: cfme/tests/cloud/test_project_quota.py [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+                             role='EvmRole-super_administrator',
+                             tenant='My Company/{}'.format(new_project.name))
+
+
+@pytest.yield_fixture(scope='module')
```
Comment (rlbabyuk) [inline cfme/tests/cloud/test_project_quota.py]: Required: please use @pytest.fixture

File: cfme/tests/cloud/test_project_quota.py [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+    user = collection.create(
+        name='user_{}'.format(fauxfactory.gen_alphanumeric()),
+        credential=new_credential,
+        email='xyz@redhat.com',
+        groups=new_group,
+        cost_center='Workload',
+        value_assign='Database')
+    yield user
+    user.delete()
+    new_group.delete()
```
Comment (rlbabyuk) [inline cfme/tests/cloud/test_project_quota.py]: Required: this should not be done in one please, as for instance group will not be deleted if user.delete() fails

File: cfme/tests/infrastructure/test_project_quota.py [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+import pytest
+
+from cfme import test_requirements
+from cfme.infrastructure.provider.rhevm import RHEVMProvider
+from cfme.infrastructure.provider.virtualcenter import VMwareProvider
+from cfme.markers.env_markers.provider import ONE_PER_TYPE
+from cfme.provisioning import do_vm_provisioning
+from cfme.utils.generators import random_vm_name
+
+from widgetastic.utils import partial_match
```
Comment (rlbabyuk) [inline cfme/tests/infrastructure/test_project_quota.py]: Required: this should be moved to the top:
import fauxfactory
import pytest
from widgetastic.utils import partial_match

File: cfme/tests/infrastructure/test_project_quota.py [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+    appliance.server.browser.refresh()
+    new_project.set_quota(**{'{}_cb'.format(field): False})
+
+
+@pytest.fixture(scope='module')
+def new_project(appliance):
+    collection = appliance.collections.projects
+    yield collection.create(name='project{}'.format(fauxfactory.gen_alphanumeric()),
+                            description='project_des{}'.format(fauxfactory.gen_alphanumeric()),
+                            parent=collection.get_root_tenant())
```
Comment (rlbabyuk) [inline cfme/tests/infrastructure/test_project_quota.py]: Required: cleanup

File: cfme/tests/infrastructure/test_project_quota.py [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+
+
+@pytest.yield_fixture(scope='module')
+def new_group(appliance, new_project):
+    collection = appliance.collections.groups
+    return collection.create(description='group_{}'.format(fauxfactory.gen_alphanumeric()),
+                             role='EvmRole-super_administrator',
+                             tenant='My Company/{}'.format(new_project.name))
```
Comment (rlbabyuk) [inline cfme/tests/infrastructure/test_project_quota.py]: Required: cleanup

File: cfme/tests/cloud/test_project_quota.py (hunk 1 above)
Comment (anurag03) [reply cfme/tests/cloud/test_project_quota.py]: I am cleaning it up in new_user fixture.

**Comment by rlbabyuk** [comment]
Will move to 1LP when PRT is ready

**Comment by rlbabyuk** [comment]
@anurag03 Please check 5.9 PRT failures

**Pull request merged**